src/NHibernate/Hql/Ast/ANTLR/Tree/BetweenOperatorNode.cs

```diff
@@ -63,25 +63,27 @@ private IASTNode GetHighOperand()
 
 		private static void Check(IASTNode check, IASTNode first, IASTNode second)
 		{
-			var expectedTypeAwareNode = check as IExpectedTypeAwareNode;
-			if (expectedTypeAwareNode != null)
+			if (!(check is IExpectedTypeAwareNode expectedTypeAwareNode) ||
+			    expectedTypeAwareNode.ExpectedType != null)
```

Comment:
So the actual change here is the `ExpectedType != null` check, the other changes are code cleanup.

Reply:
Correct.